CHANGELOG.md

```diff
@@ -7,8 +7,13 @@ The format is loosely based on [Keep a Changelog](http://keepachangelog.com/en/1
 
 ## Unreleased
 
+#### Added
+
+* `SetupSet`, `VerifySet` methods for `mock.Protected().As<>()` (@tonyhallett, #1165)
+
 #### Fixed
 
+* Virtual properties and automocking not working for `mock.Protected().As<>()` (@tonyhallett, #1165)
```

Comment:
I'm a little confused... what is this doing here all of a sudden? How is it related to the addition of `SetupSet` and `VerifySet`?

I'd prefer if we kept PRs focused on one thing at a time &mdash; this makes PRs much easier to understand and review.

Can you please remove all changes associated with this seemingly unrelated bug fix from this PR, then re-submit it as a separate PR? (I've marked the suspected changes with a &dagger; in the other review comments.)

Reply:
> or is this somehow a necessary consequence of mixing your replacer class into the existing DuckReplacer?

Indeed it is.  The automocking and virtual properties issues apply equally here.  

> Can you please remove all changes associated with this seemingly unrelated bug fix from this PR, then re-submit it as a separate PR?

They are related but to make it clear that there are bugs that apply to the current functionality - https://github.com/moq/moq4/issues/1162 then I will make a separate pull request and when that has been merged in I will proceed with this pull request.

Reply:
https://github.com/moq/moq4/pull/1185

Reply:
@tonyhallett, thanks for the confirmation, and for your patience! I've merged your other PR (great work!), so you can go ahead cleaning up this one. I'll try my best to respond more quickly this time!